### ISSUE TITLE
How to reach the goals

### ISSUE THREAD

**Issue by MarcWeber** [rollup/rollup#1438]
1) quick loading
2) full features on demand
3) only bundle code you need?
Example: public site only requires libs_public = [ .... ];   but if user clicks on "stats" it'll require libs_public_advanced = [ ... ]; after logging in user might require libs_admin = [ ... ];

So for loading speed / rendering speed you want to embed/load libs_public only, load libs_public_advanced on demand, and libs_admin only after logging in. After logging in you want to use libs_public from cache. removing unused code requires libs_public to be aware of libs_admin.

Is such design overkill? Is there any solution which comes close?

**Comment by bfred-it** [comment]
Essentially a duplicate of #372
